### PR TITLE
All: initial work on MSVC support.

### PR DESCRIPTION
**Pull request by waddlesplash**
The only reason this fails now is because of the macro expansion stuff.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,7 +1,12 @@
 cmake_minimum_required(VERSION 2.6)
 project(mGBA C)
 set(BINARY_NAME mgba CACHE INTERNAL "Name of output binaries")
-set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wall -Wextra -std=c99")
+if(NOT MSVC)
+	set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wall -Wextra -std=c99")
+else()
+	set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Dsnprintf=_snprintf -Dinline=__inline")
```
Comment (endrift) [inline CMakeLists.txt]: Please do this in src/util/common.h, as there are already some defines like this in there.

File: src/util/common.h [hunk 2]
```diff
@@ -20,6 +20,8 @@
 #include <string.h>
 
 #ifdef _MSC_VER
+#define snprintf _snprintf
+#define inline __inline
```
Comment (endrift) [inline src/util/common.h]: I'd put these a bit lower for consistency's sake.

**Comment by endrift** [comment]
Please rebase this. I removed a ton of code from the thumb instruction table that will hopefully fix most if not all of the MSVC macro issues.

**Comment by waddlesplash** [comment]
@endrift OK, rebased. The build now halts because MSVC does not have `getopt`.

**Pull request merged**